src/util/PropertyHelper.js

```diff
@@ -0,0 +1,25 @@
+import { camelToDash } from './AttributeParser';
+
+/**
+ * Helper Function to inform to call the callbacks for  watched properties and dispatch change events
+ *
+ * @param context is the elements instance
+ * @param property property that changes
+ * @param newValue
+ * @param oldValue
+ */
+export const informWatchedPropertiesAndDispatchChangeEvent = (context, property, newValue, oldValue) => {
+	// notify watched properties (after update())
+	const watch = context.watch();
+	if (property in watch) {
+		watch[property](newValue, oldValue);
+	}
+
+	// dispatch change event
+	if (
+		property in context._options['propertyOptions'] &&
+		context._options['propertyOptions'][property]['notify'] === true
+	) {
+		context.dispatch(`${camelToDash(property)}-changed`, newValue, true);
```

Comment:
@quarkus Is it always save to assume that the context will have our special dispatch method?!

Reply:
Also this will probably crash when the context is a function right?

Reply:
Thats an internal function and the assumptions to the context /this did not change at all along with the implementation.

I simply moved it to another place and made it reusable to prevent redundancies ..

Reply:
Maybe we could mention that Intention in the JsDos to prevent misuse !? 
Imho it does not make things worse if we move it out of the way here, but i could also live with limiting the store side to just call the watchers .. .